blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -65,20 +70,20 @@ All joins provided by Kafka Streams were designed with aggregation in mind, like
 - (A+C, null) <== we will never have (A+B+C)
 ```
 
-In our case, we are building a state machine and want to keep the last state of execution, meaning we do not want to see the intermediate states. In this case, we have no choice but to build a custom joiner since Kafka Streams doesn't have one built-in.
+However, we are building a state machine and want to keep the last state of execution, meaning we do not want to see the intermediate states. In this case, we have no choice but to build a custom joiner since Kafka Streams doesn't have one built-in.
 
 Our custom joiner needs to:
 - [Manually create a store](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorFromExecutionTransformer.java) that will save the last state of an execution.
 - Create a custom [merge function](https://github.com/kestra-io/kestra/blob/8a17519b5b41e8ba0ab8e0044c9c6b1e887ccd94/runner-kafka/src/main/java/io/kestra/runner/kafka/executors/ExecutorMain.java#L216-L246) that will merge the execution stream with the task results stream.
 - [Get the last value](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/streams/ExecutorJoinerTransformer.java) from the state, add the task result and emit the new state that will finally be saved on the store and the final topics.
 
-With all this, we are sure that the execution state will always be the last version, whatever the number of tasks results coming in parallel might be.
+With all this, we make sure that the execution state will always be the last version, whatever the number of tasks results coming in parallel might be.
 
-## Easy distributed workload between multiple backends
+## Distributed Workload Between Multiple Backends
 
-In [Kestra](https://github.com/kestra-io/kestra), we have a scheduler that will either look up all flows to schedule execution (timed-based) or with a long-polling mechanism (detecting files on S3 or SFTP). As we wanted to avoid a single point of failure on this service, we needed to split the flow between all instances of schedulers.
+In Kestra, a scheduler will look up all flows either with scheduled execution or with a long-polling mechanism (detecting files on S3 or SFTP). To avoid a single point of failure on this service, we needed to split the flow between all instances of schedulers.
 
-To handle that case, we rely on Kafka's consumer groups that will handle the complexity of a distributed system for us. Here's the logic:
+We rely on Kafka's consumer groups that will handle the complexity of a distributed system for us. Here's how we do it:
 - Create a [Kafka stream](https://github.com/kestra-io/kestra/blob/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/KafkaFlowListeners.java) that will read in a `KTable` and transmit all the results to a `Consumer`.
 - Listen to state change, meaning mostly `REBALANCED` streams, and empty all the flows for the `Consumer`.
```

Comment:
Is it only `REBALANCED` streams?

Reply:
In fact the code say different than `RUNNING` that will bring more status, but the notable one to mention `REBALANCED` because it's a common one that you will have very often